src/WebJobs.Script/Description/DotNet/DotNetFunctionDescriptorProvider.cs

```diff
@@ -154,6 +155,12 @@ protected override IFunctionInvoker CreateFunctionInvoker(string scriptFilePath,
                     descriptors.Add(new ParameterDescriptor(ScriptConstants.SystemExecutionContextParameterName, typeof(ExecutionContext)));
                 }
 
+                if (!descriptors.Any(p => p.Type == typeof(CancellationToken)))
+                {
+                    // Add the CancellationToken. We need this to ensure we can cancel even if the Function hasn't requested it.
```

Comment:
I realized later that we actually do still need it -- b/c we need to know when a timeout fires even if a user didn't request it in their function. So if it's not requested by the user, we need to inject it. We could also change the IFunctionInvoker interface to explicitly take a CancellationToken so we don't have to pull it out of the parameters.